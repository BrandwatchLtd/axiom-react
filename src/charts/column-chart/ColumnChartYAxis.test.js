import React from 'react';
import renderer from 'react-test-renderer';
import { ColumnChart, ColumnChartYAxis } from 'bw-axiom';

function getComponent(props = {}) {
  return renderer.create(
    <ColumnChart yAxisWidth="10rem">
      <ColumnChartYAxis { ...props }>
        Lorem
      </ColumnChartYAxis>
    </ColumnChart>
  );
}

describe('ColumnChartYAxis', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with align', () => {
    ['top', 'bottom'].forEach((align) => {
      it(align, () => {
        const component = getComponent({ align });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
