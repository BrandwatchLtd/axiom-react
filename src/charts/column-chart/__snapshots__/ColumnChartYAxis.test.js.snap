// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnChartYAxis renders with align bottom 1`] = `
<div
  className="ax-column-chart"
  style={
    Object {
      "height": undefined,
    }
  }
>
  <div
    className="ax-column-chart__y-axis ax-column-chart__y-axis--bottom"
    style={
      Object {
        "flexBasis": "10rem",
      }
    }
  >
    <div
      className="ax-column-chart__y-axis-content"
    >
      Lorem
    </div>
  </div>
</div>
`;

exports[`ColumnChartYAxis renders with align top 1`] = `
<div
  className="ax-column-chart"
  style={
    Object {
      "height": undefined,
    }
  }
>
  <div
    className="ax-column-chart__y-axis ax-column-chart__y-axis--top"
    style={
      Object {
        "flexBasis": "10rem",
      }
    }
  >
    <div
      className="ax-column-chart__y-axis-content"
    >
      Lorem
    </div>
  </div>
</div>
`;

exports[`ColumnChartYAxis renders with defaultProps 1`] = `
<div
  className="ax-column-chart"
  style={
    Object {
      "height": undefined,
    }
  }
>
  <div
    className="ax-column-chart__y-axis ax-column-chart__y-axis--bottom"
    style={
      Object {
        "flexBasis": "10rem",
      }
    }
  >
    <div
      className="ax-column-chart__y-axis-content"
    >
      Lorem
    </div>
  </div>
</div>
`;
